src/main/java/edu/ucsb/ucsbcslas/controllers/OnlineOfficeHoursController.java

```diff
@@ -99,4 +130,119 @@
+    //         String body = mapper.writeValueAsString(savedOfficeHour);
+    //         return ResponseEntity.ok().body(body);
+    //     } catch(RuntimeException e){
+    //         throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Malformed CSV", e);
+    //     }
+    // }
+
+    @PostMapping(value = "/api/admin/officehours/upload", produces = "application/json")
+    public ResponseEntity<String> uploadCSV(@RequestParam("csv") MultipartFile csv, @RequestHeader("Authorization") String authorization) throws IOException{
+        logger.info("Starting upload CSV");
+        AppUser user = authControllerAdvice.getUser(authorization);
+        if (!authControllerAdvice.getIsAdmin(authorization))
+           return getUnauthorizedResponse("admin");
+        try {
+            BufferedReader csvReader = new BufferedReader(new InputStreamReader(csv.getInputStream()));
+            logger.info(new String(csv.getInputStream().readAllBytes()));
+            
+            String body = "";
+
+            List<OnlineOfficeHours> savedOfficeHour = new ArrayList<>();
+
+            String row = csvReader.readLine();
+            String [] data ;
+            while(row != null){
+                // body += row;
+                row = row.substring(1,row.length() - 1);
+                data = row.split("\",\"");
+
+                
+                
+
+                String name = data[0];
+                String quarter = data[1];
+                String instructorFirstName = data[2];
+                String instructorLastName = data[3];
+                String instructorEmail = data[4];
+                String firstName = data[5];
+                String lastName = data[6];
+                String email = data[7];
+                String assignmentType = data[8];
+                String dayOfWeek = data[9];
+                String startTime = data[10];
+                String endTime = data[11];
+                String zoomRoomLink = data[12];
+                String notes = data[13];
+                
+                row = csvReader.readLine();
+                
+                Course course = new Course(name,quarter,instructorFirstName,instructorLastName,instructorEmail);
+                
+                Course existingCourse = courseRepository.findByNameAndQuarter(course.getName(), course.getQuarter());
+                if(existingCourse == null){
+                    courseRepository.save(course);
+                }
+                
+
+                Tutor tutor = new Tutor(firstName,lastName,email);
+                Tutor related_tutor;
+                Optional<Tutor> existingTutor = tutorRepository.findByEmail(email);
+                
+                Course related_course = courseRepository.findByNameAndQuarter(course.getName(), course.getQuarter());
+
+                if(!existingTutor.isPresent()){
```

Comment:
The logic statement in the if statement can be changed to `existingTutor.isEmpty()`. Would we be able to change to this, as this increases code readability?